web/src/stream_topic_history.ts

```diff
@@ -248,7 +257,11 @@ export class PerStreamHistory {
     }
 
     get_max_message_id(): number {
-        return this.max_message_id;
+        const unacked_message_ids_in_stream = [
+            ...get_max_msg_id_in_topics_waiting_for_ack(this.stream_id).values(),
+        ];
+        const max_message_id = Math.max(...unacked_message_ids_in_stream, this.max_message_id);
```

Comment:
Do we need a comment here -- explaining why we are not updating `this.max_message_id` ie. to keep fractional ids separated from server ids ??